### PR TITLE
Cleanup locore MIPS and RISC-V

### PR DESCRIPTION
**Pull request by qwattash**
This addresses some of the comments from https://github.com/CTSRD-CHERI/cheribsd/pull/879 back into the purecap kernel.

### CODE REVIEW

File: sys/mips/cheri/locore.S [hunk 1]
```diff
@@ -242,35 +217,10 @@ VECTOR(_locore, unknown)
...
-	/*
-	 * Process dynamic kernel relocations.
-	 *
-	 * process_kernel_dyn_relocs(code_cap, data_cap)
-	 */
-	cgetpccsetoffset	$c3, zero
-	cmove		$c4, $c28
-	ABSRELOC_LA(t0, _C_LABEL(process_kernel_dyn_relocs))
+	cgetpccsetoffset	$c4, zero
+	ABSRELOC_LA(t0, _C_LABEL(init_cap_relocs))
```
Comment (bsdjhb) [inline sys/mips/cheri/locore.S:223]: \o/

File: sys/mips/cheri/locore.S [hunk 2]
```diff
@@ -283,7 +233,7 @@ VECTOR(_locore, unknown)
 	nop
 
 	/* Clear all remaining capabilities */
```
Comment (bsdjhb) [inline sys/mips/cheri/locore.S:235]:
```suggestion
	/* Clear most remaining capabilities */
```
?

File: sys/mips/include/asm.h [hunk 3]
```diff
@@ -133,7 +133,7 @@
 	alias = sym
 
 #define	GLOBAL(sym)						\
-	.globl sym; sym:
+	.globl sym; .type sym,@object; sym:
 
 #define	ENTRY(sym)						\
 	.text; .globl sym; .ent sym; sym:
```
Comment (bsdjhb) [inline sys/mips/include/asm.h:139]: Should this have a .type of @function?

**Review by bsdjhb — APPROVED** [review]

File: sys/mips/include/asm.h (hunk 3 above)
Comment (jrtc27) [reply sys/mips/include/asm.h:139]: Implicitly done by `.ent`

**Review by bsdjhb — COMMENTED** [review]
Also, just to be sure, we copy the MIPS exception code to a new place and don't execute them in-place similar to the signal trampoline code, yes?  In that case, GLOBAL() for those instead of ENTRY() is more correct.

**Comment by qwattash** [comment]
@bsdjhb We do copy everything that was originally using `VECTOR()` macros. Perhaps I should modify the `#define` to act as GLOBAL instead of ENTRY there and leave the `VECTOR()` macros for both purecap and hybrid as it is a bit more expressive that GLOBAL for the purpose of relocatable code for trampolines.

**Comment by bsdjhb** [comment]
> @bsdjhb We do copy everything that was originally using `VECTOR()` macros. Perhaps I should modify the `#define` to act as GLOBAL instead of ENTRY there and leave the `VECTOR()` macros for both purecap and hybrid as it is a bit more expressive that GLOBAL for the purpose of relocatable code for trampolines.

I think making VECTOR() DTRT might indeed be fine, especially since you will have effectively made it unused now?

**Comment by qwattash** [comment]
Should be cleaned up now

**Review by bsdjhb — APPROVED** [review]

**Pull request merged**